### PR TITLE
nix: point to nixpkgs unstable channel for gradle 8.8

### PR DESCRIPTION
**Pull request by siddarthkay**
## Summary

This bump in `nixpkgs` is needed for `react-native` upgrade to `v0.75`
A summary of upgraded packages is below : 

- `openjdk` from `11` -> `17`
- `gradle` from `8.4` -> `8.8`
- `git` from `2.42.0` -> `2.44.1`
- `NodeJS ` from `18.19.1` -> `20.12.2`
- `Python` from `3.10.13` -> `3.10.14`
- `Clojure` from `1.11.1.1413` -> `1.11.3.1463`
- `Cocoapods` from `1.13.0` -> `1.15.2`

## Platforms
- Android
- iOS

status: ready

### CODE REVIEW

**Comment by siddarthkay** [comment]
Hey hey @status-im/mobile @status-im/wallet-mobile-devs : Please check if the changes in this PR cause any issues with your local dev environments.
Thank you!

**Comment by seanstrom** [comment]
I cleaned and built the ios app - everything seems okay so far :white_check_mark: 
I cleaned and built the andaroid app - everything seems good too :white_check_mark: 

for context I'm on macOS 13.6.7
and I'm using Xcode 15.2
and I'm using Android Studio 2023.2.1

File: nix/status-go/mobile/build.nix [hunk 1]
```diff
@@ -31,7 +31,7 @@ in buildGoPackage {
     ++ optional isAndroid openjdk
     ++ optional isIOS xcodeWrapper;
 
-  ldflags = concatStringsSep " " goBuildLdFlags;
+  ldflags = splitString " " (concatStringsSep " " goBuildLdFlags);
```
Comment (jakubgs) [inline nix/status-go/mobile/build.nix]: Eeeee, shouldn't this just be:
```suggestion
  ldflags = goBuildLdFlags;
```
Seems like you combine it then split it for no reason.

**Review by jakubgs — APPROVED** [review]
Tested locally, looks good.

**Comment by jakubgs** [comment]
Also, if all the commits here are necessary to make the upgrade work I would squash them into one commit with all the context in the description. You can combine the titles of existing commits as a list for the commit description.

**Review by ilmotta — APPROVED** [review]
Did a smoke test for NixOS, running tests, REPL usage, hot-reload, and checked the PR build, all good.

File: nix/status-go/mobile/build.nix (hunk 1 above)
Comment (siddarthkay) [reply nix/status-go/mobile/build.nix]: Thanks!

**Comment by siddarthkay** [comment]
@status-im/mobile-qa : This PR also needs just a smoke test, we upgrade all our build tools here when we move to newer `nixpkgs` version.

**Review by apentori — APPROVED** [review]

**Comment by mariia-skrypnyk** [comment]
@siddarthkay on my way to test it!

**Comment by mariia-skrypnyk** [comment]
Hey @siddarthkay !

Did smoke, all good.
PR can be merged.

**Comment by yakimant** [comment]
@siddarthkay, missed that.

Could we have gradle from unstable and everything else from stable?

**Comment by siddarthkay** [comment]
@yakimant : The easiest way to do that in existing setup could be to update our fork of `nixpkgs` from 24.05 release branch and only update gradle 8.8 in that repo and use that commit here.

**Comment by jakubgs** [comment]
@siddarthkay not true. You can have however many `nixpkgs` version as you want. It's just a variable.

**Comment by siddarthkay** [comment]
ah then what you suggest is indeed easier @jakubgs

**Comment by siddarthkay** [comment]
An attempt is here -> https://github.com/status-im/status-mobile/pull/21374

**Pull request merged**